components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx

```diff
@@ -129,6 +135,21 @@ export const PortfolioAsset = (props: Props) => {
   const coinMarketData = useUnsafeWalletSelector(WalletSelectors.coinMarketData)
   const selectedCoinMarket = useUnsafePageSelector(PageSelectors.selectedCoinMarket)
 
+  // Queries
+  const { data: isRewardsEnabled } = useGetRewardsEnabledQuery()
+  const { data: externalRewardsInfo } = useGetExternalRewardsWalletQuery()
+
+  // Constants
+  const externalRewardsProvider =
+    externalRewardsInfo?.provider ?? undefined
+
+  const rewardsToken = getRewardsBATToken(externalRewardsProvider)
+
+  const userTokensInfo =
+    isRewardsEnabled
+      ? [rewardsToken, ...userVisibleTokensInfo]
+      : userVisibleTokensInfo
```

Comment:
memoize to prevent rerenders

Reply:
can probably compute both in a single useMemo

Reply:
Memoized